### PR TITLE
Bring the data from the backend to the frontend

### PR DESCRIPTION
**Pull request by maruntav**
I apologise for the bad naming of the branch. The reviewer should be the original creator of Kodlfix, Ricardo.

### CODE REVIEW

File: src/backend/app.js [hunk 1]
```diff
@@ -1,30 +1,36 @@
...
+const fs = require("fs");
 const tvShows = require("./ShowsList");
 const path = require('path');
 const imageRes = require('./image/ResolveID');
 const port = process.env.PORT || 3001;
 
+//downloading movie data
 app.get('/rest/show', (req, res)=>{
-    
+   console.log('rest api requested');
```
Comment (rmallols) [inline src/backend/app.js]: Can you please remove the `console.log`? They shouldn't be on prod code unless you have a very good reason (e.g. if you are trying to debug an issue in prod)

File: src/backend/app.js [hunk 2]
```diff
@@ -1,30 +1,36 @@
...
+//downloading the images
+//app.use(express.static(path.join(__dirname, './image/covers')));
+app.get('/images/:id', function (req, res) {
+    let imageExists = tvShows.tvShows.find((tvShow) => tvShow.id === req.params.id);
 
-app.use(express.static(path.join(__dirname, './image/covers')));
-app.get('/images/:name', function (req, res) {
-    res.sendFile(path.join(__dirname,  './image/covers', `${req.params.name}.jpg`));
+    if(!imageExists){
+        res.redirect('/not-found');
```
Comment (rmallols) [inline src/backend/app.js]: So if an image is broken, you strictly redirect to the `not-found` page. I personally find this strategy a bit aggressive with the user!

File: src/backend/image/ResolveID.js [hunk 3]
```diff
@@ -1,2 +1,35 @@
 let cover = 1;
```
Comment (rmallols) [inline src/backend/image/ResolveID.js]: I don't really get that. So you are exporting `1`, why do you need the `coverObj`?

File: src/frontend/App.js [hunk 4]
```diff
@@ -9,18 +9,18 @@ import NotFound from './components/not-found/NotFound';
...
   
   componentDidMount(){
-    fetch('/rest/show')
-    .then((res)=> {
-      return res.json();
-    })
-    .then((myJson)=> {
-      console.log(myJson);
-    });
+    // fetch('/rest/show')
```
Comment (rmallols) [inline src/frontend/App.js]: Please remove commented code!

File: src/frontend/components/movie-details/MovieDetails.js [hunk 5]
```diff
@@ -1,59 +1,78 @@
 import React, {Component} from 'react';
 import {Redirect} from 'react-router-dom';
-import Movies from '../tv-shows/gallery/GalleryGet';
+//import Movies from '../tv-shows/gallery/GalleryGet';
```
Comment (rmallols) [inline src/frontend/components/movie-details/MovieDetails.js]: Commented code :)

File: src/frontend/components/movie-details/MovieDetails.js [hunk 6]
```diff
@@ -1,59 +1,78 @@
...
         };
     }
 
     componentWillMount(){
-        //frontend
-        let movieObject = this.findMovieObject(Movies, this.props.match.params.movie);
-        this.setState({
-            movieObject: movieObject
-        });
+       //do nothing
```
Comment (rmallols) [inline src/frontend/components/movie-details/MovieDetails.js]: If you want to do nothing, can you please just remove the method?

File: src/frontend/components/movie-details/MovieDetails.js [hunk 7]
```diff
@@ -1,59 +1,78 @@
...
+        //             backendObject: movieObj,
+        //         });
         //     });
     }
-
-    componentDidMount(){
-        //do nothing
-    }
-
     render(){
```
Comment (rmallols) [inline src/frontend/components/movie-details/MovieDetails.js]: Please remove, at least, the console.logs here. You should use devTools to debug these things :)

File: src/frontend/components/tv-shows/gallery/galleryObject.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+// let movieObj;
```
Comment (rmallols) [inline src/frontend/components/tv-shows/gallery/galleryObject.js]: More commented code :)

**Comment by maruntav** [comment]
This was sent on Monday, I thought you didn't received it. I made a new one today so you can just ignore that. Thanks

**Pull request merged**